core/src/main/scala/org/apache/spark/partial/ApproximateActionListener.scala

```diff
@@ -34,11 +34,13 @@ private[spark] class ApproximateActionListener[T, U, R](
     rdd: RDD[T],
     func: (TaskContext, Iterator[T]) => U,
     evaluator: ApproximateEvaluator[U, R],
-    timeout: Long)
+    timeout: Long = 1000*60*60*24*30*12,
```

Comment:
This submission stems from our needs that drop out the tasks which are delayed by the hardware, scheduler or the networks but not  the skew data. In a production environment,  different machines may be congested at different times. While in some machine learning algorithms like SGD, we do not  need to wait all the tasks completion in each iteration for we can guarantee the convergence as long as the slow nodes are not always slow nodes.